docs/docs/dev_docs/contracts/artifacts.md

```diff
@@ -0,0 +1,107 @@
+# Contract artifacts
+
+After compiling a contract you'll get a Contract Artifact file, that contains the data needed to interact with a specific contract, including its name, functions that can be executed, and the interface and code of those functions. Since private functions are not published in the Aztec network, you'll need this artifact file to be able to call private functions of contracts.
+
+The artifact file can be used with `aztec.js` to instantiate contract objects and interact with them.
+
+## Contract Artifact Structure
+
+The structure of a contract artifact is as follows:
+```json
+{
+  "name": "CardGame",
+  "functions": [
+    {
+      "name": "constructor",
+      "functionType": "secret",
+      "isInternal": false,
+      "parameters": [],
+      "returnTypes": [],
+      "bytecode": "...",
+      "verificationKey": "..."
+    },
+    {
+      "name": "on_card_played",
+      "functionType": "open",
+      "isInternal": true,
+      "parameters": [
+        {
+          "name": "game",
+          "type": {
+            "kind": "integer",
+            "sign": "unsigned",
+            "width": 32
+          },
+          "visibility": "private"
+        },
+        {
+          "name": "player",
+          "type": {
+            "kind": "field"
+          },
+          "visibility": "private"
+        },
+        {
+          "name": "card_as_field",
+          "type": {
+            "kind": "field"
+          },
+          "visibility": "private"
+        }
+      ],
+      "returnTypes": [
+        ...
+      ],
+      "bytecode": "...",
+      "verificationKey": "..."
+    },
+   ...
+  ]
+}
+
+```
+
+### Contract Name
+It is a simple string that matches the name that the contract developer used for this contract in noir. It's used for logs and errors.
+
+### Functions
+A contract is a collection of several functions that can be called. Each function has the following properties:
+
+#### Name
+A simple string that matches the name that the contract developer used for this function in noir. For logging and debugging purposes.
+
+#### Function Type
+The functionType can have one of the following values:
+
+- Secret: The function is ran and proved locally by the clients, and its bytecode not published to the network.
+- Open: The function is ran and proved by the sequencer, and its bytecode is published to the network.
+- Unconstrained: The function is ran locally by the clients to generate digested information useful for the user. It's not meant to be transacted against.
+
+#### Internal Flag
```

Comment:
Can we make the titles the same as the keys in the artifact? Will be easier to find the description :)